pkg/globalnet/controllers/service_controller.go

```diff
@@ -119,6 +119,11 @@ func (c *serviceController) onDelete(service *corev1.Service) (runtime.Object, b
 		return nil, false
 	}
 
+	origService := service.GetLabels()[ServiceRefLabel]
+	if origService != "" {
+		service.Name = origService
+	}
```

Comment:
So for an internal service delete this will delete the associated `GlobalIngressIP`. This doesn't seem ideal. First, the `GlobalIngressIP`"owns" the internal service and controls its lifecycle not the other way around. Second, it differs from the startup behavior where we simply clear the `GlobalIngressIP`'s allocated global IP if the internal service doesn't exist. It seems we should do the same here however I really don't think it's worth trying to handle such unexpected, erroneous cases.